code/modules/surgery/organs/stomach/external/_external_organs.dm

```diff
@@ -0,0 +1,219 @@
+	. = ..()
+
+	bodypart.external_organs.Add(src)
+	bodypart.contents.Add(src)
+
+///Use a sprite NAME (Beautiful Moth Butt Wings) and find the appropriate sprite_accessory datum
+/obj/item/organ/external/proc/add_sprite_datum(base_icon, clear_all=FALSE)
+	if(clear_all)
+		sprite_datums.Cut()
+	var/sprite_datum = get_sprite_datum(base_icon)
+	if(!sprite_datum)
+		CRASH("[type] could not find a sprite accessory datum with the display name '[base_icon]")
+
+	sprite_datums.Add(sprite_datum)
+
+///Add the overlays we need to draw on a person. Called from _bodyparts.dm
+/obj/item/organ/external/proc/get_overlays(list/overlay_list, _dir, _layer, body_type, _color)
+	generate_icon_cache()
+	for(var/datum/sprite_accessory/sprite_datum as anything in sprite_datums)
+		var/g = (body_type == FEMALE) ? "f" : "m"
+		var/finished_icon_state = (sprite_datum.gender_specific ? g : "m") + "_" + preference + "_" + sprite_datum.icon_state + mutant_bodyparts_layertext(_layer)
+		var/mutable_appearance/appearance = mutable_appearance(sprite_datum.icon, finished_icon_state, layer = -_layer)
+		appearance.dir = _dir
+
+		if(sprite_datum.color_src) //There are multiple flags, but only one is ever used so meh :/
+			appearance.color = _color
+
+		if(sprite_datum.center)
+			center_image(appearance, sprite_datum.dimension_x, sprite_datum.dimension_y)
+
+		overlay_list += appearance
+
+///Generate a unique key based on our sprites. So that if we've aleady drawn these sprites, they can be found in the cache and wont have to be drawn again (blessing and curse)
+/obj/item/organ/external/proc/generate_icon_cache()
+	cache_key = ""
+	for(var/datum/sprite_accessory/sprite as anything in sprite_datums)
+		cache_key += sprite.icon_state + "_" + preference
+
+/**This exists so sprite accessories can still be per-layer without having to include that layer's
+*  number in their sprite name, which causes issues when those numbers change.
+*/
+/obj/item/organ/external/proc/mutant_bodyparts_layertext(layer)
+	switch(layer)
+		if(EXTERNAL_BEHIND)
+			return "_BEHIND"
+		if(EXTERNAL_ADJACENT)
+			return "_ADJ"
+		if(EXTERNAL_FRONT)
+			return "_FRONT"
+
+///Because all the preferences have names like "Beautiful Sharp Snout" we need to get the sprite datum with the actual important info
+/obj/item/organ/external/proc/get_sprite_datum(sprite)
+	var/list/feature_list = get_global_feature_list()
+	return feature_list[sprite]
+
+///Return a dumb glob list for this specific feature (called from parse_sprite)
+/obj/item/organ/external/proc/get_global_feature_list()
+	return null
+
+///Check whether we can draw the overlays. You generally don't want lizard snouts to draw over an EVA suit
+/obj/item/organ/external/proc/can_draw_on_bodypart(mob/living/carbon/human/human)
+	return TRUE
+
+///The horns of a lizard!
+/obj/item/organ/external/horns
+	zone = BODY_ZONE_HEAD
+	slot = ORGAN_SLOT_EXTERNAL_HORNS
+	layers = list(EXTERNAL_ADJACENT)
```

Comment:
There's a lot of organs in the world--this is going to create a brand new list everytime.

This should just be a bitfield so that we aren't creating lists for every single organ.